ci/deploy.sh

```diff
@@ -26,8 +26,18 @@ git config --global push.default simple
 git config --global user.email "$(git log --max-count=1 --format='%ae')"
 git config --global user.name "$(git log --max-count=1 --format='%an')"
 git checkout "$BRANCH"
-git remote set-url origin "git@github.com:$REPO_SLUG.git"
 
+if [ -v SECRET_GITHUB_TOKEN ] && [ "$SECRET_GITHUB_TOKEN" != "" ]; then
+  echo >&2 "[INFO] Detected SECRET_GITHUB_TOKEN. Will deploy via HTTPS."
+  USE_GITHUB_TOKEN=true
+  git remote set-url origin "https://$SECRET_GITHUB_TOKEN@github.com/$REPO_SLUG.git"
+else
+  echo >&2 "[INFO] Missing SECRET_GITHUB_TOKEN. Will deploy via SSH."
+  USE_GITHUB_TOKEN=false
+  git remote set-url origin "git@github.com:$REPO_SLUG.git"
+fi
+
+if [ $USE_GITHUB_TOKEN = false ]; then
```

Comment:
Do you want to indent the next block?

Reply:
If we indent the following the two lines this will cause conflicts with legacy users who have modified those variable names:

```shell
  -K $encrypted_9befd6eddffe_key \
  -iv $encrypted_9befd6eddffe_iv \
```

Therefore I propose we fix the indentation when we finally remove the encrypted variable method.

Reply:
Agreed.  Let's keep it as is.